File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetUSPDataBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize

internal class GetUSPDataBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "getUSPData"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        result.success(usercentrics.instance.getUSPData().serialize())
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/GetUSPDataBridge.swift
import Usercentrics
import Flutter

struct GetUSPDataBridge : MethodBridge {

    let name: String = "getUSPData"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        result(usercentrics.shared.getUSPData().serialize())
    }
}
